AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/NetworkReachabilityNotifier.swift

```diff
@@ -16,14 +16,41 @@ struct ReachabilityUpdate {
 
 @available(iOS 13.0, *)
 class NetworkReachabilityNotifier {
+    private(set) static var shared: NetworkReachabilityNotifier?
```

Comment:
I'm just now realizing that this is singleton pattern isn't going to work if we want reachability notifiers for different APIs (e.g., one each for a GraphQL and a REST endpoint). I think we're going to need these to be straight up instances. We could consider a Factory that maintains cached reachability notifiers per host and creates or returns those instances at need.

Reply:
Thanks for pointing this out.. I will revert this.

Hmm, i think we still want the API Category to own the NetworkReachability stuff since it's closest to knowing what the endpoint is... but this means we'll either need to do something like expose a Combine publisher to the Datastore category so that our MutationRetryNotifier can subscribe (or something to that effect)

I'll dig into this.

Reply:
In the short term, I'm going to make the NetworkReachabilityNotifier as an optional -- until we figure out out how to plumb this between API and Datastore.